predicators/approaches/active_sampler_learning_approach.py

```diff
@@ -148,9 +148,10 @@ def _update_sampler_data(self) -> None:
 
     def _check_option_success(self, option: _Option, segment: Segment) -> bool:
         ground_nsrt = utils.option_to_ground_nsrt(option, self._nsrts)
-        return ground_nsrt.add_effects.issubset(
-            segment.final_atoms) and not ground_nsrt.delete_effects.issubset(
-                segment.final_atoms)
+        # Only the add effects are checked to determine option success. The
+        # philosophy is the same as pre-image backchaining (with positive
+        # preconditions).
+        return ground_nsrt.add_effects.issubset(segment.final_atoms)
```

Comment:
This isn't quite right (for instance, we could satisfy this check and still delete some important necessary atoms). I think this is fine for now, but let's update this comment to mention that this isn't really kosher in general.
I think the right thing is to call the preimage backchaining procedure to compute necessary atoms.